codice/ddf
# [2.19.x] G-8428 Refactored Logout Redirect

#### What does this PR do?
Refactored Logout Redirect to use the search UI config and apply to the search UI logout button.

#### Who is reviewing it? 
@mojogitoverhere 
@andrewzimmer 
@pklinef 

#### How should this be tested?
1.) Verify ddf behavior remains unchanged if the new config is not set:
Navigate to the Search UI
Open Chrome Dev tools or a similar function in your browser in order to see network requests
Use the signout button at the top right of the search UI to logout
Verify there is a network request with a name that starts with "main.js"

2.) Verify setting the config changes the signout button of the search UI to immediately redirect the user to the logout url instead of processing the logout action providers:
Navigate to the Admin UI
Click on the "System" tab
Click on the configuration titled "Catalog UI Search"
Verify the value for the field "Logout Url" is empty by default
Set the Logout Url to the single character "/"
Navigate to the search UI
Open Chrome Dev tools or a similar function in your browser in order to see network requests
Use the signout button at the top right of the search UI to logout
Verify there is NOT a network request with a name that starts with "main.js"
Verify you are redirected to https://localhost:8993/

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
